llvm/lib/Transforms/Utils/FlattenCFG.cpp

```diff
@@ -450,9 +450,9 @@ bool FlattenCFGOpt::MergeIfRegion(BasicBlock *BB, IRBuilder<> &Builder) {
     if (!CompareIfRegionBlock(IfTrue1, IfTrue2, SecondEntryBlock))
       return false;
   } else if (IfTrue1 == FirstEntryBlock) {
-    // The then-path is empty, so we must use "and" operation to combine the
+    // The else-path is empty, so we must use "or" operation to combine the
     // conditions.
-    CombineOp = BinaryOperator::And;
+    CombineOp = BinaryOperator::Or;
```

Comment:
Was this really just completely backwards?

Reply:
Even I had my doubts about it, but the testcase this issue had seems to give correct results on executions(output) with the changes and from my understanding the changes are correct. I'm hoping the the original author or reviewer will take a look. From my side, I'll try to regenerate all the patterns in the testcase in c file and execute and see if they're giving correct output on execution.

Reply:
The existing tests should not have the inputs changed. You should add new tests